### DISCUSSION TITLE
Memory usage on Leader/routers as network scales

### DISCUSSION THREAD

**Discussion by hakved** [openthread/openthread#9361] [Q&A] [answered]
Hi!

Is there any documentation on how much RAM is required when the network grows?

I want to calculate the theoretical maximum number of devices in a network based on memory usage,

Thanks in advanced on any help on this,
Håkon

**Comment by jwhui** [comment, marked as answer]
By design, Thread devices generally do not maintain state for all other devices in the Thread network. Routers only maintain state for directly attached children and other active routers in the network. Thread currently limits the number of active routers to 32, primarily to bound the state and communication complexity.

**Reply by hakved** [reply]
Thanks for your replay,

I got the understanding that the leader holds information about the whole network, if this is true, does the leader require more ram than the other roles?

If I understand you right, the numbers given her: https://openthread.io/guides/thread-primer/node-roles-and-types#device_limits is theoretical possible with standard radio mcu like nRF52840 with regard to ram usage?

**Reply by abtink** [reply]
Every router in network (including leader) knows about other routers in mesh. Every router can potentially take over as leader (if current leader become offline) since they all have the same information as leader. The number of active routers in mesh is limited to 32 in spec, and the memory for this is already accounted for on every router/FTD device.

Each router (parent) tracks and handles its own set of children. The other routers (or leader) do not track the children attached to other routers/parent. The number of children is limited to 511 per router/parent (by spec), but the device can be configured to limit this further (config `OPENTHREAD_CONFIG_MLE_MAX_CHILDREN`).
